### DISCUSSION TITLE
Collision system with Rapier to apply rotation?

### DISCUSSION THREAD

**Discussion by inc0** [bevyengine/bevy#4595] [Q&A]
Hello!

I'm new to bevy and gamedev in general. I'm trying to use bevy to simulate collisions of 2 very complex 3d shapes (proteins in molecular biology). I've managed to create 2 `RigidBody` from protein shape and make them collide and bounce, great! One protein is static and another is dynamic, then I'm throwing dynamic one at static and hope for it to collide and bounce, which it does.

Problem is, it bounces in exact same rotation it hit. For example, if it hits with very tip of small protruding thingy it has, it bounces away in same rotation rather than rotating around center of mass like real physical body would. Is it configurable or at all supported in Rapier?

**Comment by tim-blackbird** [comment]
In case these shapes are triangle-mesh colliders, could it be you haven't given them mass/angular inertia?
It's mentioned in the warning [here](https://rapier.rs/docs/user_guides/bevy_plugin/common_mistakes#applying-a-force-to-a-rigid-body-doesnt-work).

**Reply by inc0** [reply]
Thank you for reply! I've set up density for ColliderMassProps, I thought it should generate angular inertia properties too?
